keps/sig-windows/2578-windows-conformance/README.md

```diff
@@ -0,0 +1,434 @@
+- Ability to delete and recreate pods for StatefulSets which preserve their ability to serve as routed endpoints for services, while also having unchanging IP addresses.
+- Ability to access internal pods by internal Kubernetes DNS service endpoints.
+- Ability to access external services by Kubernetes DNS services endpoints (for non-airgapped clusters).
+- Ability to access internal pods through custom DNS records, injected by the Kubernetes pod Specification.
+- Ability to route services from pods from the EndpointSlice API.
+- Ability to run IPv4/IPv6 dual-stack networking (on supported OS Versions)
+
+#### Basic Service accounts and local storage
+
+- Ability to reboot all of the Windows nodes in a cluster while running Windows pods, in such a way that recovery of these deleted pods occurs in the scope of a few minutes.
+- Ability to delete and recreate services in such a way that load balancing rules for pods are recovered by whatever chosen service proxy is being utilized.
+- Ability to mount a hostPath storage volume from a Kubelet into a pod and write to it. 
+- Ability to mount a hostPath storage volume from a Kubelet into a pod and read from it. 
+- Ability to mount host volumes
+- Ability to rotate service accounts for running pods, with the APIServer access capacity of the pod remaining unchanged over the long term.
+- Ability to read and write shared files on a single Kubernetes node between three running Windows containers, simultaneously.
+
+#### Basic Scheduling
+
+- Ability to schedule pods with CPU and Memory limits demonstrably honored over time.
+- Ability to demonstrate the pods requesting more CPU and Memory then available are left in the Pending state over time.
+
+#### Basic concurrent functionality
+
+We intentionally do not define scale targets, as this KEP concerns itself with standardizing minimal functionality required for an operational Windows cluster.
+
+That said, some level of concurrency and repetition is required to verify that a cluster can be functional, multi-tenant workloads.
+
+Thus, we define the following simplistic "performance" tests.  Since Windows service load balancing is not as deeply hardened as that of Linux, we also propose that basic scalability requirements which limit node sizes for Windows clusters, be put in place, to ensure that "large clusters" are validated in a correct way.
+
+- Ability to reschedule a deployment Pod, 50 times in a row, with continuous deletions in the background.
+- Ability to route traffic to 10 pods behind a common ClusterIP service endpoint.
+- Ability to route traffic to 10 pods behind a common NodePort service endpoint.
+
+We do not include GMSA support in the basic definition of Windows OR because it is known to require other cluster components outside of Kubernetes's Control.
+
+### Sub-specifications of Windows Operational Readiness (Kubernetes 1.24)
```

Comment:
Should we add some checks that ensure that either Defender is disabled or if it is there are some well-known exclusions set up?
We know for sure that not including a windows defender exclusion for processes like containerd can lead to very poor performance.

I'm not sure where best to add this sort of check since it isn't really testing any K8s functionality (but i think is still important)

Reply:
```powershell
# disable certain features of microsoft defender to increase the speed of running/setting Kubernetes on Windows
Set-MpPreference -DisableRealtimeMonitoring $true -DisableScriptScanning $true -DisableArchiveScanning $true 
Set-MpPreference -ExclusionPath <BIN_PATH>

# verify our defender preferences were set
Get-MpPreference
Get-MpComputerStatus
```

These are my standard go-to commands for dealing with Defender. The exclusion path needs to be set based on what you use for your binary path(s). We can also add exclusions for specific processes, with an example below. Assuming we can use powershell to add the checks, this should be very easy to adapt.

```powershell
$processNames = @( 'containerd.exe','kubelet.exe','kube-proxy.exe','calico.exe', 'calico-node.exe', 'calico-ipam.exe' )
    ForEach ($p in $processNames) {
        Get-CimInstance -ClassName Win32_Process -Filter  "name = '$p'" | Select-Object -First 1 | ForEach-Object { 
                Set-MpPreference -ExclusionProcess $($_.Path)
        }
}
```

Reply:
I added an item as `Windows Worker Configuration`, the command can be used as part of the test, not sure if we should add in the KEP since there isn't formal examples for the other items.. @rosskirkpat